project/SparkBuild.scala

```diff
@@ -435,7 +438,19 @@ object SparkBuild extends PomBuild {
       else x.settings(Seq[Setting[_]](): _*)
     } ++ Seq[Project](OldDeps.project)
   }
+}
 
+object Core {
+  lazy val settings = Seq(
+    unmanagedResourceDirectories in Compile += baseDirectory.value / "target" / "extra-resources",
+    resourceGenerators in Compile += Def.task {
+      val buildScript = baseDirectory.value + "/../build/spark-build-info"
+      val targetDir = baseDirectory.value + "/target/extra-resources/"
+      val command =  buildScript + " " + targetDir + " " + version.value
+      Process(Seq("/bin/bash", "-c", command)).!!
```

Comment:
Same comment re: `bash -c`. Also, if users have bash somewhere else, this won't work (while the `/usr/bin/env bash` in the script would).